agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4433,49 +4433,48 @@ bool slave_thread::handle_channel_preference_query(Socket *sd, ieee1905_1::CmduM
 }
 
 /**
- * @brief check if channel is restricted in a given preference
+ * @brief Get the channel preference 
  * 
  * @param channel channel to check
- * @param preference channel preference
- * @return true if channel is restricted
- * @return false if channel is not restricted
+ * @param preference preference 
+ * @return NON_OPERABLE if channel is restricted, channel preference otherwise
  */
-static bool is_channel_restricted(const beerocks::message::sWifiChannel channel,
-                                  const son::wireless_utils::sChannelPreference &preference)
+static wfa_map::cPreferenceOperatingClasses::ePreference
+get_channel_preference(const beerocks::message::sWifiChannel channel,
+                       const son::wireless_utils::sChannelPreference &preference)
 {
     uint8_t center_channel = 0;
     auto bw                = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
     auto operating_class   = wireless_utils::get_operating_class_by_channel(channel);
 
+    // consider channels in different operating classes as restricted
+    if (operating_class != preference.oper_class) {
+        return wfa_map::cPreferenceOperatingClasses::ePreference::NON_OPERABLE;
+    }
     // operating classes 128,129,130 use center channel **unlike the other classes**,
     // so convert channel and bandwidth to center channel.
     // For more info, refer to Table E-4 in the 802.11 specification.
     if (operating_class == 128 || operating_class == 129 || operating_class == 130) {
         center_channel = wireless_utils::get_5g_center_channel(channel.channel, bw, true);
     }
 
-    // restrict channels from other operating classes
-    if (operating_class != preference.oper_class) {
-        return true;
-    }
     // According to Table 23 in the MultiAP Specification, an empty channel list field
     // indicates that the indicated preference applies to all channels in the operating class.
     if (preference.channels.empty()) {
-        return true;
+        return wfa_map::cPreferenceOperatingClasses::ePreference::NON_OPERABLE;
     }
     // explicitely restrict non-operable channels
+    auto channel_to_check =
+        (operating_class == 128 || operating_class == 129 || operating_class == 130)
+            ? center_channel
+            : channel.channel;
     for (const auto &ch : preference.channels) {
-        if (operating_class == 128 || operating_class == 129 || operating_class == 130) {
-            if (center_channel == ch.channel) {
-                return true;
-            }
-        } else {
-            if (channel.channel == ch.channel) {
-                return true;
-            }
+        if (channel_to_check == ch.channel) {
+            wfa_map::cPreferenceOperatingClasses::ePreference::NON_OPERABLE;
         }
     }
-    return false;
+    // default to the highest preference
+    return wfa_map::cPreferenceOperatingClasses::ePreference::PREFERRED14;
```

Comment:
This should actually be PREFERRED15 - but that is apparently not defined...

Reply:
This is according to spec:
```
enum ePreference {
            NON_OPERABLE = 0x0,
            PREFERRED1 = 0x1,
            PREFERRED2 = 0x2,
            PREFERRED3 = 0x3,
            PREFERRED4 = 0x4,
            PREFERRED5 = 0x5,
            PREFERRED6 = 0x6,
            PREFERRED7 = 0x7,
            PREFERRED8 = 0x8,
            PREFERRED9 = 0x9,
            PREFERRED10 = 0xa,
            PREFERRED11 = 0xb,
            PREFERRED12 = 0xc,
            PREFERRED13 = 0xd,
            PREFERRED14 = 0xe,
            RESERVED = 0xf,
        };
```

Reply:
![image](https://user-images.githubusercontent.com/13386849/76445484-24cc4100-63ce-11ea-9d1f-7ac44d970984.png)

Reply:
Exactly: if it is not explicitly specified, it should be a larger preference value than 14. So it should default to 15 (or any larger number). The spec literally says "the most preferred score 15 is inferred" (i.e. not part of the message). Internally, we still have to use that value.